utils/strings.go

```diff
@@ -79,3 +80,14 @@ func LongestCommonPrefix(list []string) string {
 	}
 	return list[0]
 }
+
+func StringHasher() func(str string) uint64 {
+	seed := maphash.MakeSeed()
+	var hash maphash.Hash
+	hash.SetSeed(seed)
+	return func(str string) uint64 {
+		hash.Reset()
+		_, _ = hash.WriteString(str)
+		return hash.Sum64()
+	}
+}
```

Comment:
Could be more efficient and simpler if we were using Go 1.19.
```go
func StringHasher() func(str string) uint64 {
	seed := maphash.MakeSeed()
	return func(str string) uint64 {
		return maphash.String(seed, str)
	}
}
```
https://pkg.go.dev/hash/maphash@master#String

Reply:
We should keep compatibility with Go 1.18 until at least 1.20 is out. We have to be able to build binaries for platforms that upgrades in a slower pace.